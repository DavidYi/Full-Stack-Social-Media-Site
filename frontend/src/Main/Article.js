import React, {useState} from 'react';
import Comment from './Comment';

import Button from 'react-bootstrap/Button';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Form from 'react-bootstrap/Form';
import FormControl from 'react-bootstrap/FormControl';

import './main.scss';

import {Formik} from 'formik';
import * as yup from 'yup';
import moment from 'moment';



function Article(props) {
	const [showResults, toggleResults] = useState(false);
	const [editBody, toggleEditBody] = useState(false);


	const schema = yup.object().shape({
		body: yup.string().required(),
	});

	const defaultValues = {
		body: props.post.body
	}

	const schemaComment = yup.object().shape({
		body: yup.string().required(),
	});

	const defaultValuesComment = {
		body: ""
	}



	const toggleComments = () => toggleResults(!showResults);

	const toggleEdit = () => toggleEditBody(!editBody);

	const handleSubmit = async (values, actions) => {
		props.editPost(values.body, props.post._id, false);
		toggleEdit();
	};

	const handleSubmitComment = async (values, actions) => {
		props.editPost(values.body, props.post._id, -1);
		actions.resetForm(defaultValues);
	};

	return (
		<div className="article">
		<Jumbotron>
			<h5>{props.post.author}</h5>
			<p className="text-muted">{moment(props.post.date).format('LLL')}</p>

			{!editBody && 
				<div> 
					<div className="">
						<p>{props.post.body}</p>
						{props.post.picture && <img src={props.post.picture} className="picture" alt="can't find"/>}
					</div>

					{props.post.author === props.username && <Button variant="outline-primary" onClick={toggleEdit}>Edit</Button>}
					<Button variant="outline-primary" onClick={toggleComments}>Comment</Button>
				</div>
			}

			{editBody && props.post.author === props.username && 
				<div>
					<Formik 
						validationSchema={schema}
						onSubmit={handleSubmit}
						initialValues={defaultValues}
					>

					{({
						handleSubmit,
						handleChange,
						handleBlur,
						handleReset,
						values,
						touched,
						isValid,
						isSubmitting,
						errors,
						resetForm
					}) => (
						<Form onSubmit={handleSubmit}>

						<FormControl 
							as="textarea"
							name="body"
							onChange={handleChange}
							value={values.body}
							isInvalid={!!errors.body} 
						/>

						<Button variant="outline-primary" type="submit" disabled={isSubmitting}>Save</Button>
						<Button variant="outline-danger" type="reset" onClick={() => {
							handleReset();
							toggleEdit();
						}}>Cancel</Button>
						</Form>
					)}
					</Formik>
				</div>
			}

			{
				showResults &&
			<div id={"article" + props.post._id}>
			<Formik 
				validationSchema={schemaComment}
				onSubmit={handleSubmitComment}
				initialValues={defaultValuesComment}
			>

			{({
				handleSubmit,
				handleChange,
				handleBlur,
				handleReset,
				values,
				touched,
				isValid,
				isSubmitting,
				errors,
				resetForm
			}) => (
				<Form onSubmit={handleSubmit}>

				<FormControl 
					className="rounded-edges"
					type="text"
					name="body"
					onChange={handleChange}
					value={values.body}
					isInvalid={!!errors.body} 
				/>
				</Form>
			)}
			</Formik>
				{props.post.comments !== undefined && props.post.comments.map((comment,index) => 
					<Comment key={"comment" + comment._id} comment={comment} username={props.username}
						post={props.post} editPost={props.editPost}/>
				)}
			</div>
			}
		</Jumbotron>
		</div>
	);
}


export default Article;