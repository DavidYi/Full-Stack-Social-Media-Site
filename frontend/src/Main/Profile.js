import React from 'react';
import {Redirect} from 'react-router-dom';
import Button from 'react-bootstrap/Button';
import Jumbotron from 'react-bootstrap/Jumbotron';
import FormControl from 'react-bootstrap/FormControl';
import Form from 'react-bootstrap/Form';

import {Formik} from 'formik';
import * as yup from 'yup';

import './main.scss';

const schema = yup.object().shape({
	status: yup.string().required(),
});

const defaultValue = {
	status: '',
};

class Profile extends React.Component {
	constructor(props) {
		super(props);
		this.props = props;
		this.state = {
			profilePressed: false,
			status: "",
			name: "",
			avatar: "",
			logOut: false,
			error: false
		};

		this.logout = this.logout.bind(this);

	}

	componentDidMount() {
		fetch("https://yo-backend-final.herokuapp.com/headline", {
				method: 'GET',
				headers: {
					'Content-Type' : 'application/json'
				},
				credentials: 'include'
			}).then(res => {
				if (!res.ok)
					throw new Error(res.error);
				return res.json();
			})
			.catch(err => this.setState({error: err}))
			.then(json => {
				this.setState({name: json.username, status: json.status})
			});

		fetch("https://yo-backend-final.herokuapp.com/avatar", {
				method: 'GET',
				headers: {
					'Content-Type' : 'application/json'
				},
				credentials: 'include'
			}).then(res => {
				if (!res.ok)
					throw new Error(res.error);
				return res.json();
			})
			.catch(err => this.setState({error: err}))
			.then(json => {
				this.setState({avatar: json.avatar})});
	}

	logout() {
		fetch("https://yo-backend-final.herokuapp.com/logout", {
				headers: {
					'Content-Type' : 'application/json'
				},
				method: 'PUT',
				credentials: 'include',
			}).then(res => this.setState({logOut: true}));
	}

	render() {
		const handleSubmit = async (values, actions) => {
			fetch("https://yo-backend-final.herokuapp.com/headline", {
				method: 'PUT',
				headers: {
					'Content-Type' : 'application/json'
				},
				credentials: 'include',
				body: JSON.stringify({ headline: values.status})
			})
			.then(res => res.json())
			.then(json => this.setState({status: json.status}));
			actions.setSubmitting(false);
			actions.resetForm();
		};

		if (this.state.error) return(<h1>{this.state.error.error}</h1>);
		return (
		<div className="profile">

		<Jumbotron className="containers">
		<Button variant="outline-danger" onClick={()=> this.logout()}>Log Out</Button>
		<Button variant="outline-info" onClick={()=>this.setState({profilePressed: true})}>Profile</Button>

		<img src={this.state.avatar} width="200px" height="260px" crop="fill" alt="loading" />
		<h1>{this.state.name}</h1>
		<p>{this.state.status}</p>
		<Formik 
			validationSchema={schema}
			onSubmit={handleSubmit}
			initialValues={defaultValue}
		>
		{({
			handleSubmit,
			handleChange,
			handleBlur,
			values,
			touched,
			isValid,
			isSubmitting,
			errors,
			resetForm
		}) => (
			<Form onSubmit={handleSubmit}>
				<FormControl 
						type="text"
						name="status"
						value={values.status}
						onChange={handleChange}
						isInvalid={!!errors.status} 
					/>
				<Button variant="outline-primary" type="submit" disabled={isSubmitting}>Update</Button>
			</Form>
		)}
		</Formik>
		</Jumbotron>
			{this.state.logOut && 
				<div>
				<Redirect to={{
		              pathname: "/"
		            }}/>
	            </div>
			}
			{this.state.profilePressed && 
				<div>
				<Redirect to={{
		              pathname: "/profile",
		              state: {
		              }
		            }}/>
	            </div>
			}
		</div>
		);
	}
}

export default Profile;