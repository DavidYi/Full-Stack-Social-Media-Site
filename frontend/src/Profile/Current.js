import React from 'react';
import {Redirect} from 'react-router-dom';
import Button from 'react-bootstrap/Button';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Form from 'react-bootstrap/Form';


import {Formik} from 'formik';
import * as yup from 'yup';

import './profile.scss';

class Current extends React.Component {
	constructor(props) {
		super(props);
		this.props = props;
		this.handleSubmit = this.handleSubmit.bind(this);
		this.state = {
			navigate: false,
			image_url: props.user.avatar,
		};

		this.defaultValue = {
			avatar: null
		};

		this.schema = yup.object().shape({
			avatar: yup.mixed().required(),
		});
	}

	async handleSubmit (values, actions) {
	    const fd = new FormData();

		fd.append('image', values.avatar);

		this.setState({loading: true});

		fetch("https://yo-backend-final.herokuapp.com/avatar", {
			method: 'PUT',
			credentials: 'include',
	    	body: fd
	    })
	    .then(res => res.json())
	    .then(json => {
	    	this.setState({image_url: json.avatar, loading: false});
			actions.setSubmitting(false);
	    });
	}

	render() {
		return (
			<div className="profile">

			<Jumbotron className="containers">
			{this.state.navigate && <Redirect to={{
				pathname: '/main'
			}}/>

			}
			<Button variant="outline-info" onClick={() => this.setState({navigate: true})}>Main Page</Button>
			<br/>
			{!this.state.loading && <img src={this.state.image_url} width="300px" height="390px" crop="fill" alt="loading" />}
			{this.state.loading && 'uploading...\n please wait a bit'}
			<Formik 
				validationSchema={this.schema}
				onSubmit={this.handleSubmit}
				initialValues={this.defaultValue}
			>
			{({
				handleSubmit,
				handleChange,
				handleBlur,
				values,
				touched,
				isValid,
				isSubmitting,
				errors,
				resetForm,
				setFieldValue
			}) => (
				<Form onSubmit={handleSubmit} noValidate>

					<div className="form-group">
						<input id="avatar" name="avatar" type="file" 
							onChange={(e) => {
								setFieldValue("avatar", e.currentTarget.files[0]);
							}} className="form-control" />
					</div>



		      		<Button variant="primary" type="submit" disabled={isSubmitting}>
		      			Upload Avatar
	      			</Button>

		      	</Form>
			)}
			</Formik>

			<h1>Current Info</h1>

			<p id="nameC">{this.props.user.username}</p>
			<p id="emailC">{this.props.user.email}</p>
			<p id="zipC">{this.props.user.zipcode}</p>
			<p id="headlineC">{new Date(parseInt(this.props.user.dob)).toString()}</p>
			
		</Jumbotron>
		</div>
		);
	}

}

export default Current;