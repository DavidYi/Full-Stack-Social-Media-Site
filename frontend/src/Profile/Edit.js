import React from 'react';
import Button from 'react-bootstrap/Button';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Form from 'react-bootstrap/Form';

import {Formik} from 'formik';
import * as yup from 'yup';


const schema = yup.object().shape({
	email: yup.string().email('Invalid email'),
	zip: yup.string().matches(/^([0-9]{5}(-[0-9]{4})?)?$/, "Invalid Zip Code"),
	password: yup.string(),
});


const defaultValue = {
	email: "",
	zip: "",
	password: ""
};


function Edit(props) {

	const handleSubmit = async (values, actions) => {

		if (values.email)
			fetch("https://yo-backend-final.herokuapp.com/email", {
				method: 'PUT',
				headers: {
					'Content-Type' : 'application/json'
				},
				credentials: 'include',
				body: JSON.stringify({ email: values.email})
			})
			.then(res => res.json())
			.then(json => {
				props.getProfile();
			});

		if (values.zip)
			fetch("https://yo-backend-final.herokuapp.com/zipcode", {
				method: 'PUT',
				headers: {
					'Content-Type' : 'application/json'
				},
				credentials: 'include',
				body: JSON.stringify({ zipcode: values.zip})
			})
			.then(res => res.json())
			.then(json => {
				props.getProfile();
			});

		if (values.password)
			fetch("https://yo-backend-final.herokuapp.com/password", {
				method: 'PUT',
				headers: {
					'Content-Type' : 'application/json'
				},
				credentials: 'include',
				body: JSON.stringify({ password: values.password})
			});
		actions.setSubmitting(false);
		actions.resetForm(defaultValue);
	};

	return (
		<div className="profile">

		<Jumbotron className="containers">
		<h1>Update Info</h1>
		<Formik 
			validationSchema={schema}
			onSubmit={handleSubmit}
			initialValues={defaultValue}
		>
		{({
			handleSubmit,
			handleChange,
			handleBlur,
			values,
			touched,
			isValid,
			isSubmitting,
			errors,
			resetForm
		}) => (
			<Form noValidate onSubmit={handleSubmit}>

	      		<Form.Group controlId="registerEmail">
					<Form.Label>Email address</Form.Label>
					<Form.Control
						type="email" 
						name="email"
						onChange={handleChange}
						isInvalid={!!errors.email}
						value={values.email}
						placeholder="Enter email"
					 />
					 <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
					 <Form.Control.Feedback type="invalid">
					 	{errors.email}
					 </Form.Control.Feedback>
		      	</Form.Group>

	      		<Form.Group controlId="registerZip">
		      		<Form.Label>Zip Code</Form.Label>
		      		<Form.Control
		      		type="text" 
		      		name="zip"
					onChange={handleChange}
					isInvalid={!!errors.zip}
					value={values.zip}
		      		placeholder="Enter Zipcode" />
		      		<Form.Control.Feedback type="invalid">
		      			{errors.zip}
		      		</Form.Control.Feedback>
	      		</Form.Group>

	      		<Form.Group controlId="registerPass">
		      		<Form.Label>Password</Form.Label>
		      		<Form.Control
		      		type="password"
		      		name="password"
					onChange={handleChange}
					isInvalid={!!errors.password}
					value={values.password}
		      		placeholder="Password" />
		      		<Form.Control.Feedback type="invalid">
		      			{errors.password}
		      		</Form.Control.Feedback>
	      		</Form.Group>

	      		<Button variant="primary" type="submit" disabled={isSubmitting}>
	      			Update Info
      			</Button>

	      	</Form>
		)}
		</Formik>
		</Jumbotron>
		</div>
	);
}


export default Edit;