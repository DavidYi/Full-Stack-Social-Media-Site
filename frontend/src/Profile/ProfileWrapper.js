import React from 'react';
import {withRouter, Redirect} from 'react-router-dom';
import Profile from './Profile';

import 'bootstrap/dist/css/bootstrap.min.css';

class ProfileWrapper extends React.Component {
	constructor(props) {
		super(props);
		this.props = props;
		this.location = this.props.location;
		this.state = {
			redirectBack: false,
			user: {},
			ready: false
		}

		this.getProfile = this.getProfile.bind(this);
	}

	getProfile() {
		fetch("https://yo-backend-final.herokuapp.com/profile",{
			method: 'GET',
			headers: {
				'Content-Type' : 'application/json'
			},
			credentials: 'include'})
			.then(res => {
				if (!res.ok){
					throw new Error(res.error);
				}
				return res.json();
			})
			.then(json => {
				this.setState({user: json, ready: true})})
			.catch(error => console.log(error));	
	}

	componentDidMount(){
		fetch("https://yo-backend-final.herokuapp.com/login", {
			headers: {
				Accept: 'application/json',
				'Content-Type' : 'application/json'
			},
			credentials: 'include',
			method: 'POST',
			body: JSON.stringify({isLoggedIn: true})
		}).then( response => response.json())
		.then(json => {
			if (!json.redirect){
				this.setState({redirectBack: true});
				return;
			}
		})

		if (this.state.redirectBack) return;

		this.getProfile();
	}

	render() {
		if (this.state.redirectBack) {
			return (<Redirect to={{
			              pathname: "/",
			            }}/>);
		}
		return (
			<div>
				{this.state.ready &&
					<Profile user={this.state.user} getProfile={this.getProfile}/>
				}
			</div>
		);
	}
}

export default withRouter(ProfileWrapper);
