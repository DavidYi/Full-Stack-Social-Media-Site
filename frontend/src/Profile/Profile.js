import React from 'react';
import Header from '../Shared/Header';
import HeaderRight from '../Shared/HeaderRight';
import Current from './Current';
import Edit from './Edit';

import Row from 'react-bootstrap/Row';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';

import 'bootstrap/dist/css/bootstrap.min.css';

function Profile(props) {
	return (
		<div>
			<Header component={<HeaderRight user={props.user}/>}/>
			<Container>
			<Row>
			<Col lg>
				<Current user={props.user} friends={props.friends} posts={props.posts}/>
			</Col>
			<Col lg>
				<Edit user={props.user} getProfile={props.getProfile}/>
			</Col>
			</Row>
			</Container>
		</div>
	);
}

export default Profile;
