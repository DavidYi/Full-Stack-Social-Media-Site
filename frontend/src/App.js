import React from 'react';
import './App.scss';
import { Route, Switch, BrowserRouter } from 'react-router-dom'
import Landing from './Landing/Landing';
import MainWrapper from './Main/MainWrapper';
import ProfileWrapper from './Profile/ProfileWrapper';
import 'bootstrap/dist/css/bootstrap.min.css';

function App() {
  return (
  	<div>

	<BrowserRouter>
		<Switch>
			<Route exact path="/" component={Landing}/>
			<Route path="/main" component={MainWrapper}/> 
			<Route path="/profile" component={ProfileWrapper}/> 
		</Switch>

	</BrowserRouter>
	</div>
  );
}

export default App;
