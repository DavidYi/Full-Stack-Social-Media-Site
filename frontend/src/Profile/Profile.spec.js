import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Profile from './Profile';
import ProfileWrapper from './ProfileWrapper';
import Current from './Current';
import Edit from './Edit';
import Article from '../Main/Article';

import App from '../App';

import {Redirect} from "react-router-dom";

import { shape } from 'prop-types';
import { BrowserRouter } from 'react-router-dom';
import Alert from 'react-bootstrap/Alert';


const routerEmpty = {
  history: new BrowserRouter().history,
  route: {
    location: {
    	state: {
    		form : {},
    		friends: []
    	},
    },
    match: {},
  },
};


const createContext = (router) => ({
  context: { router },
  childContextTypes: { router: shape({}) },
});

export function mountWrap(node, router=routerEmpty) {
  return mount(node, createContext(routerEmpty));
}

export function shallowWrap(node, router=routerEmpty) {
  return shallow(node, createContext(routerEmpty));
}

describe('Profile', () => {

	test('snapshot renders', () => {
		const component = shallowWrap(<ProfileWrapper />);
	    expect(component).toMatchSnapshot();
	});

});

async function getwrapper() {
	let form;
	let postsList = [];
	let posts = [];
	let comments = [];
	let users;
	await fetch('https://jsonplaceholder.typicode.com/users')
		.then(response => response.json())
		.then(json => {
			users = [...json];
			let u = json[0];
			form = {
				id: u.id,
				name: u.username,
				email: u.email,
				phone: u.phone,
				zip: u.address.zipcode,
				status: u.company.catchPhrase,
				password: u.address.street,
				img: 'https://uwosh.edu/deanofstudents/wp-content/uploads/sites/156/2019/02/profile-default.jpg',
			};
			comments.push({
				user: form,
				body: "congrats!",
			})
		});
	await fetch('https://jsonplaceholder.typicode.com/posts')
		.then(response => response.json())
		.then(json => {
			json.forEach(post => {
				if (post.userId === form.id) {
					post['date'] = (new Date()).toString();
					post['img'] = "https://image.shutterstock.com/image-photo/red-bunny-rabbit-portrait-looking-260nw-589399619.jpg";
					postsList.unshift(post);
					posts.unshift(<Article key={"article" + post.id}
					 user={form} post={post}
					 comments={comments}/>);
				}
			})
		});

	const wrapper = mountWrap(<Profile posts={posts} postsInfo={postsList} 
				user={form} users={users} 
				friends={[]} nextId={200}
				comments={comments}/>);

	return {wrapper: wrapper, form: form,postsList: postsList, posts: posts,comments: comments,users: users
	};
}

describe('Main wrapped', () => {

	it("check if page has current info", async () => {
		const stuff = await getwrapper();
		const wrapper  = stuff.wrapper;
		const user = stuff.form;

		const currentComp = wrapper.find(Current);
		const name = currentComp.find('p[id="nameC"]').at(0);
		const email = currentComp.find('p[id="emailC"]').at(0);
		const phone = currentComp.find('p[id="phoneC"]').at(0);
		const zip = currentComp.find('p[id="zipC"]').at(0);

		expect(name.text()).toEqual(user.name);
		expect(email.text()).toEqual(user.email);
		expect(phone.text()).toEqual(user.phone);
		expect(zip.text()).toEqual(user.zip);
	});

});

// 	const container = mount(<App />)
// 	test('snapshot renders', () => {
// 		const component = renderer.create(<Landing />);
// 	    let tree = component.toJSON();
// 	    expect(tree).toMatchSnapshot();
// 	});

// 	it('should have 2 email fields', () => {
//     	expect(container.find('input[type="email"]').length).toEqual(2);
//   	});

//   	it('should have 3 password fields', () => {
//     	expect(container.find('input[type="password"]').length).toEqual(3);
//   	});

//   	it('should have 2 forms', () => {
//     	expect(container.find('form').length).toEqual(2);
//   	});

// });


// describe('Register', () => {
// 	const container = mount(shallow(<Register />).get(0));

// 	it('renders the inner LogIn Component', () => {
// 		const wrapper = mount(<App />);
// 		expect(wrapper.find(Register).length).toEqual(1);
// 	});

// 	it('should have email field', () => {
//     	expect(container.find('input[type="email"]').length).toEqual(1);
//   	});

//   	it('should have display name field', () => {
//     	expect(container.find('input[name="name"]').length).toEqual(1);
//   	});

//   	it('should have zipcode field', () => {
//     	expect(container.find('input[name="zip"]').length).toEqual(1);
//   	});

//   	it('should have phone field', () => {
//     	expect(container.find('input[name="phone"]').length).toEqual(1);
//   	});


//   	it('should have 2 password fields (password and passcmf)', () => {
//     	expect(container.find('input[type="password"]').length).toEqual(2);
//     	expect(container.find('input[name="password"]').length).toEqual(1);
//     	expect(container.find('input[name="passCmf"]').length).toEqual(1);
//   	});

//   	it('should have 1 submit button', () => {
//     	expect(container.find('button').length).toEqual(1);
//   	});



// 	it("renders alert that registration is successful when registration is all right otherwise nothing", () => {
// 		const wrapper  = mount(shallow(<LogIn />).get(0));

// 		const email = wrapper.find('input').at(0);
// 		const name = wrapper.find('input').at(1);
// 		const phone = wrapper.find('input').at(2);
// 		const zip = wrapper.find('input').at(3);
// 		const password = wrapper.find('input[name="password"]');
// 		const passCmf = wrapper.find('input[name="passCmf"]');
// 		const submit = wrapper.find('button');
// 		email.instance().value = 'davidcyyi@rice.edu';
// 		password.instance().value = '1';
// 		passCmf.value = '12';

// 		submit.simulate('click');

// 		setTimeout(() => {
// 		expect(wrapper.find(Alert).length).toEqual(0);}, 1000);


// 		name.instance().value = 'hey';
// 		phone.value = "222-111-1111";
// 		zip.value = "12345";

// 		setTimeout(() => {
// 		expect(wrapper.find(Alert).length).toEqual(0);}, 1000);

// 		passCmf.value = '1';


// 		setTimeout(() => {expect(wrapper.find(Redirect).length).toEqual(0);
// 		expect(wrapper.find(Alert).length).toEqual(1);}, 1000);



// 	});

// 	it("should alert if wrong credentials", () => {
// 		const wrapper  = mount(shallow(<LogIn />).get(0));

// 		const email = wrapper.find('input').at(0);
// 		const password = wrapper.find('input').at(1);
// 		const submit = wrapper.find('button');
// 		email.instance().value = 'Sincere@april.biz';
// 		password.instance().value = 'Kulas Light1';

// 		submit.simulate('click');


// 		setTimeout(() => {expect(wrapper.find(Alert)).toBeTruthy();
// 		expect(wrapper.find(Alert).length).toEqual(1);}, 1000);
// 	});

// });

// describe('LogIn', () => {
// 	const container = mount(shallow(<LogIn />).get(0));

// 	it('renders the inner LogIn Component', () => {
// 		const wrapper = mount(<App />);
// 		expect(wrapper.find(LogIn).length).toEqual(1);
// 	});

// 	it('should have email field', () => {
//     	expect(container.find('input[type="email"]').length).toEqual(1);
//   	});

//   	it('should have password field', () => {
//     	expect(container.find('input[type="password"]').length).toEqual(1);
//   	});

//   	it('should have 1 submit button', () => {
//     	expect(container.find('button').length).toEqual(1);
//   	});


// 	it("renders Redirect when user autheticated", () => {
// 		const wrapper  = mount(shallow(<LogIn />).get(0));

// 		const email = wrapper.find('input').at(0);
// 		const password = wrapper.find('input').at(1);
// 		const submit = wrapper.find('button');
// 		email.instance().value = 'Sincere@april.biz';
// 		password.instance().value = 'Kulas Light';

// 		submit.simulate('click');


// 		setTimeout(() => {expect(wrapper.find(Redirect)).toBeTruthy();
// 		expect(wrapper.find(Redirect).length).toEqual(1);}, 1000);
// 	});

// 	it("should alert if wrong credentials", () => {
// 		const wrapper  = mount(shallow(<LogIn />).get(0));

// 		const email = wrapper.find('input').at(0);
// 		const password = wrapper.find('input').at(1);
// 		const submit = wrapper.find('button');
// 		email.instance().value = 'Sincere@april.biz';
// 		password.instance().value = 'Kulas Light1';

// 		submit.simulate('click');


// 		setTimeout(() => {expect(wrapper.find(Alert)).toBeTruthy();
// 		expect(wrapper.find(Alert).length).toEqual(1);}, 1000);
// 	});

// });