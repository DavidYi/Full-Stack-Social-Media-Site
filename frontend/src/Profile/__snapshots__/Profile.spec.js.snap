// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapshot renders 1`] = `ShallowWrapper {}`;
